src/dns.rs

```diff
@@ -0,0 +1,41 @@
+use no_std_net::IpAddr;
+use heapless::{String, consts};
+
+/// This is the host address type to be returned by `gethostbyname`.
+///
+/// An IPv4 address type always looks for `A` records, while IPv6 address type
+/// will look for `AAAA` records
+#[derive(Clone, Debug, PartialEq)]
+pub enum AddrType {
```

Comment:
Is there anything in `no_std_net` we can use here. Feels like we're re-inventing the wheel?

Reply:
Also, should we support an 'Either' option, in case you support both and don't mind either? I think you can ask a DNS sever for both AAAA and A records at the same time, so this would be more efficient than asking for one then the other.

Reply:
I had sort of the same feeling, but they don't define anything like this. All they have is the `IpAddr` and `SocketAddr`, and the variants in both hold `Ipv4Addr/Ipv6Addr` and `SocketAddrV4/SocketAddrV6`.

> Also, should we support an 'Either' option, in case you support both and don't mind either? I think you can ask a DNS sever for both AAAA and A records at the same time, so this would be more efficient than asking for one then the other.

I think this makes sense. An alternative could be to change `addr_type` in `gethostbyname` to `Option<AddrType>`? Where `None` actually means either. Not sure which is better?

Reply:
@thejpster Did you have any opinions on adding an `Either` vs changing it to an `Option`? See above comment.

Reply:
Sorry, I'm not on Github much any more. I think I prefer `Either|IPv6|IPv4` over `Some(IPv4|IPv6)|None`, as `None` sounds you like don't want either, as opposed to both.